src/srcset.js

```diff
@@ -197,10 +197,10 @@ export class Srcset {
    * @private
    */
   selectByWidth_(width, dpr) {
-    let minIndex = -1;
-    let minWidth = 1000000;
-    let minScore = 1000000;
-    for (let i = 0; i < this.sources_.length; i++) {
+    const length = this.sources_.length;
+    const minWidth = this.sources_[length - 1].width / dpr;
```

Comment:
With our sorting algorithm, we can't guarantee that this value isn't the `undefined`:

```js
[{ width: 640 }, { width: 320 }, { width: undefined}].sort(sortByWidth);
// => [{ width: 640 }, { width: 320 }, { width: undefined}]
```

Actually, the `undefined` width doesn't move at all when sorting, so we can't guarantee it's in any position.